python/ModelTools.py

```diff
@@ -53,6 +53,7 @@ def doSet(self,name,vars):
         if self.options.bin: self.out.defineSet(name,vars)
         else: self.out.write("%s = set(%s);\n" % (name,vars));
     def doObj(self,name,type,X):
+        if self.out.obj(name): return 1 # Do not try to reproduce constraint term if it exists
```

Comment:
We can hold off on the merge for now...

On Fri, May 15, 2015 at 5:01 PM, Giovanni Petrucciani <
notifications@github.com> wrote:

> In python/ModelTools.py
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/192#discussion_r30415714>
> :
>
> > @@ -53,6 +53,7 @@ def doSet(self,name,vars):
> >          if self.options.bin: self.out.defineSet(name,vars)
> >          else: self.out.write("%s = set(%s);\n" % (name,vars));
> >      def doObj(self,name,type,X):
> > +        if self.out.obj(name): return 1 # Do not try to reproduce constraint term if it exists
>
> I'd rather not change this, since at the moment it can trap other possible
> mistakes of object redefinition.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/192/files#r30415714>
> .
>